### PR TITLE
Turn the WebSocket server into a real Mojolicious application

### PR DESCRIPTION
**Pull request by kraih**
These are preparations for replacing dbus RPC in the WebSocket server with HTTP endpoints (Redis is not actually needed). `OpenQA::WebSockets::Server` used to be a mess of spaghetti code around a `Mojolicious::Lite` core, making it really hard to test and change things. Now it is a normal `Mojolicious` application, with controllers, models, and plugins, very similar to `OpenQA::WebAPI`. 

This is only the first step though. The class `OpenQA::WebSockets::Server` will soon vanish, and be renamed to `OpenQA::WebSockets` (which currently contains the dbus glue code). And the `ws_*` functions available via dbus RPC will be replaced with HTTP endpoints. Authentication is already prepared and tested for this.

Progress: https://progress.opensuse.org/issues/46778

### CODE REVIEW

**Comment by kraih** [comment]
Hope this isn't too much of a "huge code drop", i've already split up the refactoring from adding new features.

File: lib/OpenQA/WebSockets/Controller/Auth.pm [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+    my $schema  = OpenQA::Schema->singleton;
+    my $api_key = $schema->resultset('ApiKeys')->find({key => $key});
+    if ($api_key) {
+        if (time - $timestamp <= 300) {
+            my $exp = $api_key->t_expiration;
+            # It has no expiration date or it's in the future
+            if (!$exp || $exp->epoch > time) {
+                if (my $secret = $api_key->secret) {
+                    my $sum = hmac_sha1_sum($self->req->url->to_string . $timestamp, $secret);
```
Comment (coolo) [inline lib/OpenQA/WebSockets/Controller/Auth.pm]: no way to avoid duplicating this?

File: t/lib/OpenQA/Test/Utils.pm [hunk 2]
```diff
@@ -187,17 +188,23 @@ sub create_websocket_server {
...
         $ENV{MOJO_LISTEN}             = "http://127.0.0.1:$port";
         $ENV{MOJO_INACTIVITY_TIMEOUT} = 9999;
+
         use OpenQA::WebSockets;
         use Mojo::Util 'monkey_patch';
         use OpenQA::WebSockets::Server;
+        use OpenQA::WebSockets::Controller::Worker;
+        use OpenQA::WebSockets::Plugin::Helpers;
+
+        # TODO: Kill it with fire!
```
Comment (kalikiana) [inline t/lib/OpenQA/Test/Utils.pm]: Can you perhaps add a hint as to what you'd like to see here? You've mentioned it before in IRC but such comments quickly lose context

File: lib/OpenQA/WebSockets/Controller/Auth.pm (hunk 1 above)
Comment (kraih) [reply lib/OpenQA/WebSockets/Controller/Auth.pm]: I just moved the code here from the old `Mojolicious::Lite` app (no changes to the logic have been made). I do see some other uses of `hmac_sha1_sum` in `OpenQA::WebAPI`, but no copypasta. I guess it might make sense to look into sharing more code between WebAPI and WebSockets in a followup PR. If you spot more such cases please point them out with a comment.

File: t/api/02-iso.t [hunk 3]
```diff
@@ -31,7 +31,6 @@ use Test::Warnings;
 use OpenQA::Test::Case;
 use OpenQA::Client;
 use Mojo::IOLoop;
-use Data::Dump;
```
Comment (kalikiana) [inline t/api/02-iso.t]: There's a lot of noise from `use` changes involving `Dumper` which seems unrelated 🤔️

Comment (kraih) [reply t/api/02-iso.t]: Yes, that's not directly related. I just noticed that there's **a lot** of dead imports and removed the ones i encountered.

File: t/lib/OpenQA/Test/Utils.pm (hunk 2 above)
Comment (kraih) [reply t/lib/OpenQA/Test/Utils.pm]: Unfortunately i don't have a solution for this specific case yet. It is different from the other uses of mocked functions/methods, because it hidden deep inside a test module and there is no indication of what is happening from the API side. So there's some very dangerous action at a distance stuff going on here.

File: lib/OpenQA/WebSockets/Controller/Auth.pm (hunk 1 above)
Comment (coolo) [reply lib/OpenQA/WebSockets/Controller/Auth.pm]: I see. github folding the deletes and showing the additions fooled me.
